ios/Classes/SwiftBackgroundLocationTrackerPlugin.swift

```diff
@@ -35,6 +41,56 @@ extension SwiftBackgroundLocationTrackerPlugin: FlutterPlugin {
         locationManager.delegate = self
         SwiftBackgroundLocationTrackerPlugin.foregroundChannel?.handle(call, result: result)
     }
+    
+    public static func getFlutterEngine()-> FlutterEngine? {
+        if (flutterEngine == nil) {
+            let flutterEngine = FlutterEngine(name: flutterThreadLabelPrefix, project: nil, allowHeadlessExecution: true)
+            
+            guard let callbackHandle = SharedPrefsUtil.getCallbackHandle(),
+                  let flutterCallbackInformation = FlutterCallbackCache.lookupCallbackInformation(callbackHandle) else {
+                CustomLogger.log(message: "No flutter callback cache ...")
+                return nil
+            }
+            flutterEngine.run(withEntrypoint: flutterCallbackInformation.callbackName, libraryURI: flutterCallbackInformation.callbackLibraryPath)
+            SwiftBackgroundLocationTrackerPlugin.flutterPluginRegistrantCallback?(flutterEngine)
+            self.flutterEngine = flutterEngine
+        }
+        return flutterEngine
+    }
+    
+    public static func initBackgroundMethodChannel(flutterEngine: FlutterEngine) {
+        if (backgroundMethodChannel == nil) {
+            CustomLogger.log(message: "backgroundMethodChannel is nil, we will create a new background method channel.")
+            backgroundMethodChannel = FlutterMethodChannel(name: SwiftBackgroundLocationTrackerPlugin.BACKGROUND_CHANNEL_NAME, binaryMessenger: flutterEngine.binaryMessenger)
+            backgroundMethodChannel!.setMethodCallHandler { (call, result) in
```

Comment:
It's better to unwrap the optional with a if let or guard let, now the app can crash when backgroundMethodChannel is nil for some reason.